ui/apps/platform/src/Containers/Clusters/ClustersPage.tsx

```diff
@@ -32,7 +32,7 @@ function ClustersPage(): ReactElement {
 
     // Handle changes to the currently selected deployment.
     const setSelectedClusterId = useCallback(
-        (newCluster) => {
+        (newCluster: any) => {
```

Comment:
I noticed that you had to add `: any` in several files.
Curious why TS type inference no longer works with React 18?

Reply:
these were actually done by a react 18 typedef codemod (https://github.com/eps1lon/types-react-codemod) -- i ended up having to downgrade the typedef back to 17 because of breaking changes and dependency conflicts (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210)

Reply:
This is probably a positive change, definitely better than the implicit `any`s we were getting before. At least this way it is apparent that type safety is off in these functions. It might be worth a follow up ticket to go through and add the correct types to these functions.

I know at one point we discussed turning on the `noImplicitAny` tsconfig rule, but I think we decided against it short term due to the number of changes that were required.